sdk/keyvault/keyvault-keys/src/index.ts

```diff
@@ -419,6 +420,31 @@ export class KeyClient {
     });
   }
 
+  /**
+   * Gets a {@link CryptographyClient} for the given key.
+   *
+   * Example usage:
+   * ```ts
+   * let client = new KeyClient(url, credentials);
+   * // get a cryptography client for a given key
+   * let cryptographyClient = client.getCryptographyClient("MyKey");
+   * ```
+   * @param name - The name of the key used to perform cryptographic operations.
+   * @param version - Optional version of the key used to perform cryptographic operations.
+   * @returns - A {@link CryptographyClient} using the same options, credentials, and http client as this {@link KeyClient}
+   */
+  public getCryptographyClient(name: string, version?: string): CryptographyClient {
+    // The goals of this method are discoverability and performance (by sharing a client and pipeline).
+    // The existing cryptography client does not accept a pipeline as an argument, nor does it expose it
+    // so we _fake_ it by constructing it with the keyUrl, a fake token (which will be thrown away), and then
+    // set the remote provider's generated client to our current client.
+    const keyUrl = new URL(["keys", name, version].filter(Boolean).join("/"), this.vaultUrl);
+    const cryptoClient = new CryptographyClient(keyUrl.toString(), {} as TokenCredential);
+    // We know it has a remote provider, since this is a remoteable crypto client
+    cryptoClient["remoteProvider"]!["client"] = this.client;
```

Comment:
wait why this syntax here? :0

Reply:
Oh, now I know! — let’s add a comment on why this is this way :P no sneaking around.

Reply:
❤️ I wrote a pretty long comment here https://github.com/Azure/azure-sdk-for-js/pull/17529/files/2005755646322288ffee75440d79bd4fadce9bdf#diff-97994b12cb1d5e2e291577395e4bf975b701ab26dadfcfa8990f88900f0825f5R437-R440 is it still unclear after reading it?

Reply:
Well, it could be just for me , but the fact that each line has a comment kind of goes around how much the private part of the comment is related to the weird `["syntax"]!`. Anyhow, after the changelog entry, I’ll approve.

Reply:
That's fair - here's what I did:
1. Cleaned up the terse syntax using a conditional instead of !
2. Updated the comment to specify that we're accessing a private variable and why. 
is this better?